e2e/upgrades/upgrades_test.go

```diff
@@ -100,6 +100,59 @@ func TestUpgradesMinorVersion(t *testing.T) {
 	steps.Run(t)
 }
 
+// TestUpgradesMinorVersion tests a minor version bump
+func TestUpgradesMajorVersion20to21WithCockroachDbVersion(t *testing.T) {
+
+	// We are testing a Minor Version Upgrade with
+	// partition update
+	// Going from v20.2.10 to v21.1.0
+
+	if parallel {
+		t.Parallel()
+	}
+	if testing.Short() {
+		t.Skip("skipping test in short mode.")
+	}
+
+	testLog := zapr.NewLogger(zaptest.NewLogger(t))
+
+	actor.Log = testLog
+
+	sb := testenv.NewDiffingSandbox(t, env)
+	sb.StartManager(t, controller.InitClusterReconcilerWithLogger(testLog))
+	os.Setenv("RELATED_IMAGE_COCKROACH_v21_1_1", "cockroachdb/cockroach@sha256:7c84559a33db90b52f8179c904818525e45852b683bd6272f61dcf54c103f5b1")
+	os.Setenv("RELATED_IMAGE_COCKROACH_v20_2_10", "cockroachdb/cockroach@sha256:a1ef571ff3b47b395084d2f29abbc7706be36a826a618a794697d90a03615ada")
+	builder := testutil.NewBuilder("crdb").WithNodeCount(3).WithTLS().
```

Comment:
Can this just be a 1 node cluster? We're testing that the upgrade happens, not that the rolling restart functionality works right?

Reply:
@udnay actually it is a 3 nodes cluster... I already fixed it. It was a copy of the test for upgrades using the image field... I fixed it also there. The description of the test says 1 node but it is a 3 nodes cluster.

Reply:
Right, so what is this test actually trying to validate? The image string?

Reply:
This test should validate that we can upgrade a cluster using the cockroachDBversion field instead of the image field, using the supported versions env var that we declare in the RELATED_IMAGES env. vars. Actually it is the scenario that Keith described in the issue.  Without the fix I added this test would finish with a broken upgrade.

Reply:
Can you make that more explicit in the test itself so people understand the intent of this test?

Reply:
I added some more comments, but please check again and tell me if you need also other changes